### ISSUE TITLE
Timeout when mounted by Managed Identity 

### ISSUE THREAD

**Issue by jeffwmiles** [Azure/azure-storage-fuse#388]
### Which version of the blobfuse was used?
1.2.2 built from source

### Which OS (please include version) are you using?
`OS: Oracle Linux Server 7.7 x86_64`
`Kernel: 4.14.35-1902.10.8.el7uek.x86_64`

### What problem was encountered?
I can **successfully use blobfuse with authType Key**. However, I'm trying to use Managed Identity. 

I have created a system managed identity for an Azure virtual machine. This MI is given RBAC as "Storage Blob Data Contributor" over a storage account. 
 When I use my mount command (see below), it mounts successfully, according to output within /var/log/messages:
```
Apr 24 13:23:53 train-t-orc1 kernel: fuse init (API version 7.27)
Apr 24 13:23:53 train-t-orc1 systemd: Mounting FUSE Control File System...
Apr 24 13:23:53 train-t-orc1 systemd: Mounted FUSE Control File System.
```

I can see the volume mounted with df -h:
```
Filesystem                      Size  Used Avail Use% Mounted on
/dev/sdb1                        16G  8.1G  6.8G  55% /mnt/resource
tmpfs                           792M     0  792M   0% /run/user/1000
blobfuse                         16G  8.1G  6.8G  55% /mnt/stgacct
```

However, if I try to interact with the mounted storage in any way (save a file, list contents, etc) the command prompt gets stuck with no response or additional output. I have to restart my SSH session in order to gain access back to the VM. 

Using a connection.cfg file produces the same results. 

### Have you found a mitigation/solution?
Not while using Managed Identity.

### By default, blobfuse logs errors to syslog. If this is relevant, is there anything in the syslog that might be helpful?
This is output from the /var/log/messages after letting a "ls /mnt/<blob>" sit for 20 minutes:
[https://gist.github.com/jeffwmiles/05ffe0f6081135d9c190f1ea363d0077](https://gist.github.com/jeffwmiles/05ffe0f6081135d9c190f1ea363d0077) 

### If relevant, please share your mount command.
```
export AZURE_STORAGE_ACCOUNT=stgacct
export AZURE_STORAGE_AUTH_TYPE=MSI
./blobfuse /mnt/stgacct--tmp-path=/mnt/resource/blobfusetmp --container-name=fuse -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 --log-level=LOG_DEBUG --file-cache-timeout-in-seconds=120 -o allow_other
```

**Comment by NaraVen** [comment]
Could you please try our latest bits? We fixed an MSI related performance issue.

**Comment by jeffwmiles** [comment]
@NaraVen I've re-built from master and am still seeing the exact same behavior, including the log output I included in my gist.

**Comment by NaraVen** [comment]
I did not see errors in the attached log file. How much memory/cpu do you have in that machine?

**Comment by jeffwmiles** [comment]
Its 2 vCPU and 8 GB RAM (B2ms size). 

Line 30 of the gist states:
`list_blobs_hierarchical failed for the 1 time with errno = 35`
after multiple GETs, with many seconds in between each attempt.

**Comment by NaraVen** [comment]
Looks like a memory issue in the machine. What s the size of the largest file and what is the total storage size?

**Comment by jeffwmiles** [comment]
The storage account has 2 files in it - a 22 Mb executable and a 30 Kb png file. I'm attempting to do this as proof of concept and am keeping it simple. 
However, after mounting when I do a `df -h` the space used shows 8.1 GB
![df](https://user-images.githubusercontent.com/40004712/80991844-e3687880-8df5-11ea-9898-3da4d6938452.png)

While the list command is stuck in its processing, I'm monitoring memory usage of processes, and there's lots available:
![top](https://user-images.githubusercontent.com/40004712/80991873-ed8a7700-8df5-11ea-8d5b-a12e203b14a2.png)

Here is the comparison when I mount the same container with an access key instead of Managed Identity (no real difference in df or top)
![df_key](https://user-images.githubusercontent.com/40004712/80992346-9933c700-8df6-11ea-9025-49265987f4c3.png)

![ls](https://user-images.githubusercontent.com/40004712/80992357-9d5fe480-8df6-11ea-819a-f9c524acdd4f.png)

![top_key](https://user-images.githubusercontent.com/40004712/80992371-a3ee5c00-8df6-11ea-95d7-38b573c1af4f.png)

**Comment by NaraVen** [comment]
what do you use for MSI endpoint , default Azure or a custom endpoint. are you using user_identity or system identity?.

**Comment by jeffwmiles** [comment]
It is a default Azure System identity:
![mi](https://user-images.githubusercontent.com/40004712/81073742-511fad80-8ea5-11ea-8c7a-0c885575c9d0.png)

**Comment by NaraVen** [comment]
Could you please try our latest code. Managed Identity will be slower that a storage account Key since token refrieval and refresh is required on expiry.

**Comment by jeffwmiles** [comment]
No change - using blobfuse 1.2.3, either trying to do an 'ls' or 'cp' command to storage mounted with system Managed Identity. Terminal just sits without output for over 5 minutes for either command.

**Comment by vibhansa-msft** [comment]
We have done some performance and memory enhancements in version 1.3.3. Kindly try that out and let us know if the issue still persists.

**Comment by NaraVen** [comment]
we have released 1.3.4 and tested MSI is faster than before. So, I am closing this one feel free to reopen.
